modules/OCPP201/conversions.cpp

```diff
@@ -819,6 +819,70 @@ to_ocpp_get_15118_certificate_request(const types::iso15118::RequestExiStreamSch
     return _request;
 }
 
+ocpp::v2::EnergyTransferModeEnum to_ocpp_energy_transfer_mode(const types::iso15118::EnergyTransferMode transfer_mode) {
+    switch (transfer_mode) {
+    case types::iso15118::EnergyTransferMode::AC_single_phase_core:
+        return ocpp::v2::EnergyTransferModeEnum::AC_single_phase;
+    case types::iso15118::EnergyTransferMode::AC_two_phase:
+        return ocpp::v2::EnergyTransferModeEnum::AC_two_phase;
+    case types::iso15118::EnergyTransferMode::AC_three_phase_core:
+        return ocpp::v2::EnergyTransferModeEnum::AC_three_phase;
+    case types::iso15118::EnergyTransferMode::DC:
+    case types::iso15118::EnergyTransferMode::DC_core:
+    case types::iso15118::EnergyTransferMode::DC_extended:
+    case types::iso15118::EnergyTransferMode::DC_combo_core:
+    case types::iso15118::EnergyTransferMode::DC_unique:
+        return ocpp::v2::EnergyTransferModeEnum::DC;
+
+    case types::iso15118::EnergyTransferMode::AC_BPT:
+    case types::iso15118::EnergyTransferMode::AC_BPT_DER:
+    case types::iso15118::EnergyTransferMode::AC_DER:
+    case types::iso15118::EnergyTransferMode::DC_BPT:
+    case types::iso15118::EnergyTransferMode::DC_ACDP:
+    case types::iso15118::EnergyTransferMode::DC_ACDP_BPT:
+    case types::iso15118::EnergyTransferMode::WPT:
+        throw std::out_of_range("Could not convert EnergyTransferModeEnum: OCPP does not know this type");
+    }
+
+    throw std::out_of_range("Could not convert EnergyTransferMode");
+}
+
+ocpp::v2::NotifyEVChargingNeedsRequest
+to_ocpp_notify_ev_charging_needs_request(const types::iso15118::ChargingNeeds& charging_needs) {
+    ocpp::v2::NotifyEVChargingNeedsRequest _request;
+    ocpp::v2::ChargingNeeds& _charging_needs = _request.chargingNeeds;
+
+    // TODO(ioan): missing 'evseId'?
+    _request.evseId = 1;
```

Comment:
Every connected iso15118_extension requires a mapping to an EVSE. Have a look how this is done for the energy sinks: https://github.com/EVerest/everest-core/blob/main/lib/staging/external_energy_limits/external_energy_limits.cpp#L25

If no mapping is configured, we can not map it and therefore can not send the respective message.

Reply:
Please implement this in a similar way and add the module mapping to all configs where OCPP is included